src/Symfony/Bridge/Doctrine/Messenger/DoctrineClearEntityManagerWorkerSubscriber.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Messenger;
+
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Messenger\Event\WorkerMessageFailedEvent;
+use Symfony\Component\Messenger\Event\WorkerMessageHandledEvent;
+
+/**
+ * Clears entity managers between messages being handled to avoid outdated data.
+ *
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+class DoctrineClearEntityManagerWorkerSubscriber implements EventSubscriberInterface
+{
+    private $managerRegistry;
+
+    public function __construct(ManagerRegistry $managerRegistry)
+    {
+        $this->managerRegistry = $managerRegistry;
+    }
+
+    public function onWorkerMessageHandled()
+    {
+        $this->clearEntityManagers();
+    }
+
+    public function onWorkerMessageFailed()
+    {
+        $this->clearEntityManagers();
+    }
+
+    public static function getSubscribedEvents()
```

Comment:
could be simplified to clear before handling with just `WorkerMessageReceivedEvent`

Reply:
You're totally right. What I don't know is: is doing it before vs after significant? I was following the "lead" of the middleware... but I'm not sure if it matters or not.

Reply:
"after" will probably free up memory sooner, in the case they no messages come in for a while.

Reply:
The fact that I need to listen to 2 events relates to #32614, which attempts to fix that.